### PR TITLE
Tonk.approve site image

### LINKED ISSUE #203: Add a new route to an admin to approve a site image
**Why**
- in part of site image uploading, we want admins to be able to approve appropriate images of tree

**What to do**
- implement a new route that is called to approve the image
- See [the API reference](https://docs.c4cneu.com/sftt/sftt-api-spec/#approve-site-image-admin-only) for more details

### PR DESCRIPTION
**Pull request by CerberusLatrans**
## Why
Resolves #203 
Allows admins to approve site images with a specified image id.

## This PR
Changes the approval status of the image with the given id to `APPROVED` regardless of the previous status.
Credit to @nalinikantheti for doing all the programming

## Verification Steps
Tested on postman to make sure that uploaded images switch to `APPROVED`

### CODE REVIEW

File: service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java [hunk 1]
```diff
@@ -184,34 +183,45 @@ record -> {
...
     // if no approved images exist for the entry, check if its tree has a default image.
-    // to counter any minor differences in tree name (e.g. Honey locust vs Honeylocust), 
-    // normalize the tree's common name by setting the name to all lowercase and removing empty spaces
-    String defaultUrl = db.select(TREE_SPECIES.DEFAULT_IMAGE).from(TREE_SPECIES)
-        .where(lower(replace(TREE_SPECIES.COMMON_NAME, " ", ""))
-            .eq(commonName.toLowerCase().replace(" ", "")))
-        .fetchOne(0, String.class);
+    // to counter any minor differences in tree name (e.g. Honey locust vs Honeylocust),
+    // normalize the tree's common name by setting the name to all lowercase and removing empty
+    // spaces
```
Comment (huang0h) [inline service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java:217]: auto-formatter is not kind to comments 😞

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java [hunk 2]
```diff
@@ -861,14 +862,26 @@ public void editSiteEntry(JWTData userData, int entryId, UpdateSiteRequest editS
...
+  }
+
+  @Override
+  public void approveSiteImage(JWTData userData, int imageID) {
+    assertAdminOrSuperAdmin(userData.getPrivilegeLevel());
+    checkImageExists(imageID);
+    SiteImagesRecord imageRecord =
+        db.selectFrom(SITE_IMAGES).where(SITE_IMAGES.ID.eq(imageID)).fetchOne();
+    imageRecord.setApprovalStatus(ImageApprovalStatus.APPROVED.getApprovalStatus());
+    imageRecord.store();
```
Comment (huang0h) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:885]: 💯

**Review by huang0h — APPROVED** [review]
Awesome work!

File: api/src/main/java/com/codeforcommunity/dto/site/SiteEntry.java [hunk 3]
```diff
@@ -61,6 +61,7 @@ public class SiteEntry {
       pattern = "MM/dd/yyyy",
```
Comment (SurabhiKeesara) [inline api/src/main/java/com/codeforcommunity/dto/site/SiteEntry.java:61]: This is nit picky, but try to remove these line changes

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java (hunk 2 above)
Comment (SurabhiKeesara) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:885]: Does this just store/push these changes to the database?

**Review by SurabhiKeesara — APPROVED** [review]
This looks great. I would just suggest cleaning up some of the formatting changes where you aren't actually changing the code (unless those are necessary style fixes).

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java (hunk 2 above)
Comment (CerberusLatrans) [reply service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:885]: Yes

**Pull request merged**